src/ast/passes/semantic_analyser.cpp

```diff
@@ -2819,7 +2819,15 @@ void SemanticAnalyser::visit(AssignMapStatement &assignment)
 
   const std::string &map_ident = assignment.map->ident;
 
-  if (type.IsRecordTy() && map_val_[map_ident].IsRecordTy()) {
+  if (type.IsCastableMapTy() && map_val_[map_ident].IsIntTy()) {
+    if (dynamic_cast<Call *>(assignment.expr)) {
+      LOG(ERROR, assignment.map->loc, err_)
```

Comment:
Unit tests for the error case would also be good

Reply:
@ajor See my resolved comment above. These are already covered by a semantic analyser test.

Reply:
Got confused by the error message staying the same!